### PR TITLE
Add Unit Tests for Gauss-Chebyshev Quadratures

### LINKED ISSUE #24: Add Unit Test for GaussChebyshev2
#22 fixed the GaussChebyshev2 implementation, but it still needs a unit test.

### PR DESCRIPTION
**Pull request by wavefunction91**
Fixes bugs (e.g. https://github.com/wavefunction91/IntegratorXX/pull/27#issuecomment-1626295323) and adds unit tests for Gauss-Chebyshev quadratures. Closes #24.

### CODE REVIEW

File: include/integratorxx/quadratures/gausscheby3.hpp [hunk 1]
```diff
@@ -62,14 +62,14 @@ struct quadrature_traits<GaussChebyshev3<PointType, WeightType>> {
 
     weight_container weights(npts);
     point_container points(npts);
-    for (size_t i = 1; i <= npts; ++i) {
+    for (size_t i = 0; i < npts; ++i) {
       // The standard nodes and weights are given by
-      points[i] = std::cos(0.5 * (2*i - 1) * pi_ov_2n_p_1);
+      points[i] = std::pow(std::cos(0.5 * (2*(i+1) - 1) * pi_ov_2n_p_1),2);
```
Comment (wavefunction91) [inline include/integratorxx/quadratures/gausscheby3.hpp:67]: @susilehtola Would like a second pair of eyes here - based on the AS equations, this is the proper expression

File: include/integratorxx/quadratures/gausscheby3.hpp [hunk 2]
```diff
@@ -62,14 +62,14 @@ struct quadrature_traits<GaussChebyshev3<PointType, WeightType>> {
...
       weights[i] = 2.0*pi_ov_2n_p_1 * points[i];
 
       // However, since we want the rule with a unit weight factor, we
-      // divide the weights by sqrt((1-x)/x).
-      weights[i] /= std::sqrt((1.0 - points[i])/points[i]);
+      // divide the weights by sqrt(x/(1-x)).
+      weights[i] *= std::sqrt((1.0 - points[i])/points[i]);
```
[the remark this answers is not in the document]
Comment (susilehtola) [reply include/integratorxx/quadratures/gausscheby3.hpp:72]: Yeah, it was correct modulo multiplication / division :joy:

File: include/integratorxx/quadratures/gausscheby3.hpp (hunk 1 above)
Comment (susilehtola) [reply include/integratorxx/quadratures/gausscheby3.hpp:67]: Yeah, this looks more correct

**Review by susilehtola — APPROVED** [review]

File: include/integratorxx/quadratures/gausscheby1.hpp [hunk 3]
```diff
@@ -64,16 +65,18 @@ struct quadrature_traits<GaussChebyshev1<PointType, WeightType>> {
...
-      points[i] = std::cos((2.0 * (i + 1) - 1.) / two_npts_x_pi);
-      weights[i] = pi_ov_npts;
+      points[i] = -std::cos(ti);
+      weights[i] = pi_ov_npts; 
       
       // However, as we're integrating f(x) not \frac{f(x)}{\sqrt{1 -
       // x^2}}, we must factor the \sqrt{1-x^2} into the weight
-      weights[i] *= std::sqrt(1. - std::pow(points[i],2));
+      // sqrt(1 - cos^2(x)) = abs(sin(x))
+      weights[i] *= std::sin(ti); //std::sqrt(1. - std::pow(points[i],2));
```
Comment (susilehtola) [inline include/integratorxx/quadratures/gausscheby1.hpp:79]: I am not sure about this change: computing the square root is way cheaper than computing `sin`!

File: include/integratorxx/quadratures/gausscheby1.hpp [hunk 4]
```diff
@@ -64,16 +65,18 @@ struct quadrature_traits<GaussChebyshev1<PointType, WeightType>> {
...
     const weight_type pi_ov_npts = M_PI / npts;
-    const weight_type two_npts_x_pi = 2 * npts * M_PI;
+    const weight_type pi_ov_2npts = pi_ov_npts / 2;
 
     for (size_t i = 0; i < npts; ++i) {
+      const auto ti = (2.0 * (i + 1) - 1.) * pi_ov_2npts;
       // The standard nodes and weights are given by
-      points[i] = std::cos((2.0 * (i + 1) - 1.) / two_npts_x_pi);
-      weights[i] = pi_ov_npts;
+      points[i] = -std::cos(ti);
```
Comment (susilehtola) [inline include/integratorxx/quadratures/gausscheby1.hpp:73]: Why the minus sign?!

Comment (wavefunction91) [reply include/integratorxx/quadratures/gausscheby1.hpp:73]: This is ensures that the quadrature is generated in ascending order, which is important for ensuring the radial transforms are in ascending order (which is the format needed to interop with the pruner for the spherical quadratures) - I believe Psi4 generated in descending order, potato potato.

File: include/integratorxx/quadratures/gausscheby1.hpp (hunk 3 above)
Comment (wavefunction91) [reply include/integratorxx/quadratures/gausscheby1.hpp:79]: Good call, will fix.

File: include/integratorxx/quadratures/gausscheby1.hpp (hunk 4 above)
Comment (wavefunction91) [reply include/integratorxx/quadratures/gausscheby1.hpp]: On second thought, this should be done via indexing ala #34, changes back for now

**Pull request merged**